### PR TITLE
Nil pointer dereference panic fix 

### PR DESCRIPTION
**Pull request by iulianpascalau**
- started the health service as the first step when creating a node
- fixed one integration test
- added mutex protection in interceptors
- added more metrics in the trie syncer

### CODE REVIEW

File: trie/statistics/trieSyncStatistics.go [hunk 1]
```diff
@@ -110,6 +131,22 @@ func (tss *trieSyncStatistics) NumTries() int {
 	return len(tss.missingMap)
 }
 
+// ProcessorTime will return the cumulative processor time used in tries synchronization (sum of all go routines used in trie sync processes)
+func (tss *trieSyncStatistics) ProcessorTime() time.Duration {
```
Comment (bogdan-rosianu) [inline trie/statistics/trieSyncStatistics.go:135]: might rename `ProcessorTime` / `AddProcessorTime` to `ProcessingTime` / `AddProcessingTime`

Comment (bogdan-rosianu) [reply trie/statistics/trieSyncStatistics.go:135]: or `CumulatedTime` / `AddToCumulatedTime`

File: trie/statistics/trieSyncStatistics_test.go [hunk 2]
```diff
@@ -106,3 +107,37 @@ func TestTrieSyncStatistics_BytesReceived(t *testing.T) {
...
+	t.Parallel()
+
+	tss := NewTrieSyncStatistics()
+
+	assert.Equal(t, time.Duration(0), tss.ProcessorTime())
+
+	tss.AddProcessorTime(time.Second)
+	assert.Equal(t, time.Second, tss.ProcessorTime())
+
+	tss.AddProcessorTime(time.Millisecond)
```
Comment (bogdan-rosianu) [inline trie/statistics/trieSyncStatistics_test.go:138]: can you update this test so it uses more goroutines? same for the above one

File: trie/statistics/trieSyncStatistics.go (hunk 1 above)
Comment (iulianpascalau) [reply trie/statistics/trieSyncStatistics.go:135]: done

File: trie/statistics/trieSyncStatistics_test.go (hunk 2 above)
Comment (iulianpascalau) [reply trie/statistics/trieSyncStatistics_test.go:138]: added

**Review by bogdan-rosianu — APPROVED** [review]

**Review by raduchis — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Pull request merged**